rpc/handlers.go

```diff
@@ -151,6 +151,13 @@ func (h *Handler) BlockWithTxHashes(id BlockID) (*BlockWithTxHashes, *jsonrpc.Er
 	}, nil
 }
 
+func (h *Handler) LegacyBlockWithTxHashes(id BlockID) (*BlockWithTxHashes, *jsonrpc.Error) {
+	block, err := h.BlockWithTxHashes(id)
+	block.BlockHeader.L1GasPrice = nil
+	block.BlockHeader.StarknetVersion = ""
```

Comment:
Why do we need explicit assignments here? It's going to zero valued by default

Reply:
they are set by the `h.BlockWithTxHashes`, we need to clear them as those fields are not present in RPC v0.4.0.